lib/pdf_fill/extras_generator_v2.rb

```diff
@@ -369,37 +370,43 @@ def add_text(value, metadata)
       metadata[:format_options] ||= {}
       metadata[:format_options][:label_width] ||= @default_label_width
 
-      value = apply_humanization(value, metadata[:format_options])
-
-      question_num = metadata[:question_num]
-      if @questions[question_num].blank?
-        question_text = @question_key[question_num]
-        @questions[question_num] = get_question(question_text, metadata)
+      config = find_question_config(metadata)
+      if config && @questions[config[:question_number]].blank?
+        @questions[config[:question_number]] = get_question(config[:question_text], metadata, config)
```

Comment:
`config` will never be nil here, so you could remove that first line in `get_question`.